editor/src/clj/editor/code/view.clj

```diff
@@ -2485,11 +2505,34 @@
           (when (and (< drawn-line-index drawn-line-count)
                      (< source-line-index source-line-count))
             (let [y (data/row->y layout source-line-index)]
-              (when (contains? breakpoint-rows source-line-index)
+              (when (contains? breakpoint-row->condition source-line-index)
                 (.setFill gc gutter-breakpoint-color)
                 (.fillOval gc
-                           (+ (.x line-numbers-rect) (.w line-numbers-rect) 3.0)
-                           (+ y 3.0) indicator-diameter indicator-diameter))
+                           (+ (.x line-numbers-rect) (.w line-numbers-rect) indicator-offset)
+                           (+ y indicator-offset) indicator-diameter indicator-diameter)
+                (when (breakpoint-row->condition source-line-index)
```

Comment:
This lookup could maybe be merged with the `contains?` check above so we don't traverse the map twice?